### PR TITLE
add resource parameters and iterators

### PR DESCRIPTION
**Pull request by pfouque**
add functions to iterate resources
add possibility to pass custom parameters to resource functions

### CODE REVIEW

**Comment by browniebroke** [comment]
@pfouque feel free to remove Python 3.2 from Travis as it's not going to be supported from the end of 2016.

**Comment by pfouque** [comment]
Ok, I'll do that. Thanks
By the way, do you want me to add some tests concerning new features ?
Another question, i plan to implement some new entrypoints like charts APIs, I presume you prefer i do it in another PR, but how to manage if i want to create iterators?

**Comment by browniebroke** [comment]
Python 3.2 is not part of the build matrix anymore on master.

File: deezer/resources.py [hunk 1]
```diff
@@ -55,6 +55,26 @@ def get_relation(self, relation, **kwargs):
...
                                       self, **kwargs)
 
+    def iter_relation(self, relation, **kwargs):
+        """
+        Generic method to iterate relation from any resource.
+
+        Query the client with the object's known parameters
+        and try to retrieve the provided relation type. This
+        is not meant to be used directly by a client, it's more
+        a helper method for the child objects.
```
Comment (browniebroke) [inline deezer/resources.py]: To reflect the docstring into the code could you change the name of the method to start by '_' as per the python conventions.

**Comment by browniebroke** [comment]
Thanks for your contribution @pfouque 👍

**Comment by pfouque** [comment]
You're welcome!

See you soon :)

**Pull request merged**